doc/Yang Model Guidelines.md

```diff
@@ -0,0 +1,253 @@
+#   YANG MODEL GUIDELINES
+
+### Rev 1.0 ###
+
+### Revision ###
+
+ | Rev |     Date    |       Author       | Change Description                |
+ |:---:|:-----------:|:------------------:|-----------------------------------|
+ | 1.0 | 22 Aug 2019             |      Praveen Chaudhary      | Initial version                   |
+
+
+
+## This document lists the guidelines, which will be used to write Yang Modules for Sonic. These Yang Modules will be primarily based on ABNF Schema of Sonic. 
```

Comment:
May be we should just mention that RFC 7950 should be base reference point for writing SONiC YANG model.